notes/2025-xx-xx-insecurities-remedies-i.md

```diff
@@ -0,0 +1,410 @@
+- [NixOS Wiki on Systemd Hardening](https://wiki.nixos.org/wiki/Systemd/Hardening)
+- [General tips on Systemd Hardening](https://github.com/alegrey91/systemd-service-hardening)
+- [K4YT3X's Hardened & Optimized Linux Kernel Parameters](https://github.com/k4yt3x/sysctl/blob/master/sysctl.conf)
+- [Notes on SSH Hardening](https://www.sshaudit.com/hardening_guides.html)
+- [Hardened Profile module in Nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/profiles/hardened.nix)
+
+## The Problem
+
+Systemd, as a central part of many modern Linux distributions, provides various
+utilities designed to streamline system management. However, its centralization
+can also mean a single point of failure, and its wide array of features offers a
+large attack surface if misconfigured. One of the utilities it offers, which
+will come in very handy today, is **systemd-analyze security**.
+
+Go ahead and run `sudo systemd-analyze security` on your system. You will notice
+_very quickly_ that it contains a lot of "UNSAFE" and "EXPOSED" services. First
+of all, do not worry. Your system is _probably safe_. The assessment done by
+Systemd, as I will emphasize time and time again, are **entirely arbitrary**.
+They are rule-based score calculations based on your configuration, do not
+correspond to or reflect upon the actual vulnerability of your system; there is
+more nuance to such vulnerabilities than a rule-based analysis tool can
+determine. Indeed, the score assigned to each service is _not_ an indicator of
+how secure or insecure it is. It is, however, a start. In the case of Systemd,
+hardening services is a second line of defense when the executable the service
+is for becomes the vulnerability.
+
+## The Solution
+
+`systemd-analyze security <unit>` generates a score for a given unit, showing
+all the used directives. Based on limited information we have, it is possible to
+try and harden individual services.
+
+By default, Systemd leaves running services in the open. This is understandable,
+as trying to preemptively harden a service is likely to cause conflicts with
+individual services and their requirements.
+
+Although NixOS makes _some_ effort to harden services, installed services--be it
+through NixOS' services or your inferior distribution's package manager--will be
+running with little to no hardening - which you can confirm by viewing the
+security report. This post, as per my experience, aims to detail potential
+hardening options. Throughout this post, please keep those following core
+principles in mind:
+
+1. There is **no** "one size fits all" kind of hardening, all services will
+   require your undivided attention to make sure everything continues to work as
+   intended.
+2. No kind of hardening can catch all kinds of exploits. The key to a secure
+   system is to remain ever-vigilant.
+
+> Some hardening options will disable access to certain paths, or make them
```

Comment:
I'd cite where the quote came from for further reading and accreditation.

Reply:
It's not a quote, I said that.

Reply:
Oh, maybe I'd mention where you said it something. I think that was unclear.

Reply:
I think it's pretty fucking clear that I've said this when it's on my blog in a post that I've written. Don't you think?

Reply:
Usually `>` quotes are for things that other people have said. Traditionally in English grammar, you wouldn't even quote at all even if it's something you said in the past. But I kind of think that's a little bs because of publishing and such. Though, should at least note from where.

Reply:
But yeah, usually you'll leave quotes to excerpts from another source such as book, website, or article. And then cite where it was found. If it's original thought, it isn't a quote. If it's something you said in another source, you'd still cite that as well.

Reply:
Purdue Owl kind of leads the standard for MLA amd such style writing. You can look at how they describe quotes [here] (https://owl.purdue.edu/owl/general_writing/punctuation/quotation_marks/index.html).

I know this isn't probably MLA style writing. But you won't find a more reliable source for English language writing that writing standards like APA or MLA. So can kind of apply what they're saying to `> ` markdown quotes as it's more of just a different syntax.

Reply:
Hmm, that's an interesting point. But I don't think it's clear that it is being used like a github alert tbf. I know I'm probably really getting into the nitty gritty here but would it be possible to somehow signal that like how github alerts do?

Reply:
I do not feel inclined to, and excuse me for being crass, give a shit about what Purdue Owl thinks when I'm writing my own blog. This is not an academic writeup, and has never claimed to be. I believe it is clear enough, and has not been a problem for anyone who is not trying extra hard to be pedantic. If you think you can style those closer to github's alerts in generated HTML, I'll welcome a PR. I tried in the past, and couldn't quite get it right.

"But it's confusing"

I disagree, move on.

Reply:
I might give you a PR then. I think I managed to do it in the past with HTML.

Reply:
You're probably going for like "Note", "Tip", or "Important", yeah?

Reply:
And sorry. I sent the Purdue Owl thing before I saw your next reply. Probably seemed out of place after your next response.

Reply:
Either way, this is kind of non-urgent and probably shouldn't stop you from merging this PR. Can always just ammend the alert changes after.

Reply:
Should I start an issue to move this out of the PR?

Reply:
Yes an issue would be great, I'm looking into how alerts are converted and I might have a better idea than filters (or inline HTML, but I have already expressed how much I am against inline HTML.)